requests-handler.ts

```diff
@@ -0,0 +1,133 @@
+import {DefaultContext, ExtendableContext} from 'koa';
+import {
+    retrieveAllProducts,
+    retrieveSpecificProduct,
+    createProduct,
+    updateProduct,
+    deleteProduct,
+    checkout
+} from './db-actions';
+import {IRouterParamContext} from 'koa-router';
+import {creationSchema, updateSchema, checkoutSchema} from './request-schemas';
+import {logger} from './logger';
+
+const sendAllProductsRequest = async (ctx: ExtendableContext & IRouterParamContext) => {
+    logger.info(`Got GET - get all products request from ${ctx.request.ip}`);
+    try {
+        ctx.body = await retrieveAllProducts();
+    } catch (err) {
+        ctx.status = 500;
+        ctx.body = 'Internal Server Error';
```

Comment:
in development environment you want to return the error to user - in production just internal server error. you can do it in error handler middleware

Reply:
done (for 500 error status)